torch/distributed/_distributed_c10d.py

```diff
@@ -85,17 +85,23 @@
     from torch.distributed._C_stubs import HashStore
 
 # NVSHMEM/SymmetricMemory components
+
+# There are multiple backends for SymmetricMemory, as a result,
+# _SymmetricMemory should not be imported together with NVSHMEM related modules.
+try:
+    from torch._C._distributed_c10d import _SymmetricMemory
+except ImportError:
+    from torch.distributed._C_stubs import _SymmetricMemory
```

Comment:
should we always do this, to uncouple import from nvshmem even if it's available, and import nvshmem explicitly when needed?